### PR TITLE
Docs - logging best practice

### PR DESCRIPTION
**Pull request by reyang**
Add the initial doc for logging best practices.
Updated the logging benchmark to reflect the latest recommendations.

### CODE REVIEW

**Review by cijothomas — APPROVED** [review]

File: docs/logs/README.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+Refer to the [logging performance
+benchmark](../../test/Benchmarks/Logs/LogBenchmarks.cs) for more details.
+
+## Package Version
+
+:heavy_check_mark: You should always use the
+[`ILogger`](https://docs.microsoft.com/dotnet/api/microsoft.extensions.logging.ilogger)
+interface from the latest stable version of
+[Microsoft.Extensions.Logging](https://www.nuget.org/packages/Microsoft.Extensions.Logging/)
+package, regardless of the .NET runtime version being used:
```
Comment (CodeBlanch) [inline docs/logs/README.md:45]: Might be good to explicitly bless the usage of `ILogger<T>` here as well? Don't want people to think they shouldn't use it 🤷

You should always use the [`ILogger`](https://docs.microsoft.com/dotnet/api/microsoft.extensions.logging.ilogger) or [`ILogger<TCategoryName>`](https://docs.microsoft.com/dotnet/api/microsoft.extensions.logging.ilogger-1) interfaces...

Comment (reyang) [reply docs/logs/README.md:45]: Next PR. I plan to cover how to retrieve the logger (DI, LoggerFactory) and lifecycle management.

File: docs/logs/README.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+* If you're using the latest stable version of [OpenTelemetry .NET
+  SDK](../../src/OpenTelemetry/README.md), you don't have to worry about the
+  version of `Microsoft.Extensions.Logging` package because it is already taken
+  care of for you via [package dependency](../../Directory.Packages.props).
+* Starting from version `3.1.0`, the .NET runtime team is holding a high bar for
+  backward compatibility on `Microsoft.Extensions.Logging` even during major
+  version bumps, so compatibility is not a concern here.
```
Comment (CodeBlanch) [inline docs/logs/README.md:53]: I think "high bar" might not be clear to everyone. Maybe something like...

* Starting from version `3.1.0` the .NET runtime team has committed to preserve backwards compatibility on the `Microsoft.Extensions.Logging` packages. There will be no breaking changes (within reason) across major version bumps.

Comment (reyang) [reply docs/logs/README.md:53]: I don't think we can put such strong statement https://github.com/open-telemetry/opentelemetry-dotnet/pull/4920#discussion_r1346923479.

**Pull request merged**